DEVELOPMENT.md

```diff
@@ -379,6 +379,177 @@ Comments can also be used to indicate steps to take in the future (*TODOs*). Suc
+
+### Code Comments
+
+Additionally to the overall code comment rules outline above, we use [JSDoc](https://jsdoc.app) conform documentation of classes and functions. We also use full english sentences and ending sentence periods here.
+
+Please make sure that these multi-line comments follow the correct format, especially leaving the first line of this multiline JSDoc comments empty:
+
+```js
+// Bad.
+
+/** Converts a byte string from hex to base64.
+ * @param {string} bytes - The bytes, represented as hex string.
+ * @returns {string} The bytes, represented as base64 string.
+ */
+
+// Good.
+
+/**
+ * Converts a byte string from hex to base64.
+ * @param {string} bytes - The bytes, represented as hex string.
+ * @returns {string} The bytes, represented as base64 string.
+ */
+```
+
+It also makes sense to wrap code bits, variable names and URLs in \`\`  quotes, so they can easily be recognized and do not clash with our capitalization rules enforced by eslint, when they are at the beginning of a sentence:
+
+```js
+// Bad. This will get flagged by the linter.
+
+// devAddr is a hex string.
+const devAddr = '270000FF'
+
+// Good.
+
+// `devAddr` is a hex string.
+const devAddr = '270000FF'
+```
+
+### Import Statement Order
+
+Our `import` statements use the following order, separated by emty newlines:
+
+1. Node "builtin" modules (e.g. `path`)
+2. External modules (e.g. `react`)
+3. Well-known modules (e.g. `../../../components/*`)
+  1. Constants modules 
+  2. API module
+  3. Presentational components (local and global)
+  4. Container components (local and global)
+  5. View components
+  6. Utility components
+  7. Utility modules
+  8. Store modules
+  9. Global assets and styles
+4. Other parent modules (e.g. `../../../module`)
+5. Other sibling modules (e.g. `./validation-schema`, `./button.styl`)
+6. Index of the current directory (`./`)
+
+Note that this order is enforced by our linter and will cause a CI fail when not respected. Again, settting up your development environment to integrate linting will assist you greatly here.
+
+### React Component Syntax (Functional, Class Components and Hooks)
+
+We currently do not enforce a ruleset that enforces a specific component type. Lately, we have been embracing React Hooks and write a lot of new components using this approach. We use conventional class components mostly to make use of class decorators, which we have been using from the very start. Likewise, there are a lot of class components from the time before React Hooks.
+
+- Functional components should be considered:
+  - To make use of React's Hook API
+  - For plain (dumb) presentational components
+  - For components that don't need to be decorated (`@decorator`)
```

Comment:
Yeah, this is an acknowledgment to the fact that we're still using a lot of HOCs and wrapper functions and I guess it will stay that way for some time to come unless we do a heavy refactor. To me, it's still fine to use decorators, mainly because I really dislike concatenating wrapper functions when using functional components.